<img style="float: left; margin: 30px 15px 15px 15px;" src="https://upload.wikimedia.org/wikipedia/commons/d/db/Logo_ITESO_normal.jpg" width="300" height="500" /> 
    
    
### Simulación de procesos financieros. 

**Nombres:** `[Jesús Guillermo Sainz Zárate, Eduardo Gutiérrez Flores]`

**Fecha:** 21 de febrero del 2024.

**Expediente** : `[7530501,741305]`.
**Profesor:** Oscar David Jaramillo Zuluaga.
    
**Link Github**: https://github.com/GuillermoSainz07/ProyectoConjunto_EGutierrez_GSainz

# Tarea 4: Clase 8


# Ejercicio 1
Como ejemplo simple de una simulación de Monte Carlo, considere calcular la probabilidad de una suma particular del lanzamiento de tres dados (cada dado tiene valores del uno al seis). Además cada dado tiene las siguientes carácterísticas: el primer dado no está cargado (distribución uniforme todos son equiprobables); el segundo y tercer dado están cargados basados en una distribución binomial con parámetros (`n=5, p=0.5` y `n=5, p=0.2`). Calcule la probabilidad de que la suma resultante sea 7, 14 o 18. 

# Ejercicio 2
En un casino, un jugador está jugando a lanzar un dado justo de 6 caras. Si el jugador lanza un número par, el casino paga $\$1$ al jugador, de lo contrario, el jugador paga $\$1$ al casino. El jugador decide jugar 100 veces.

1. ¿Cuál es la probabilidad de que un jugador gane dinero después de 100 lanzamientos del dado bajo las reglas del juego (el casino paga $\$1$ si el resultado es par y el jugador paga $\$1$ si es impar)?

2. ¿Cómo afectaría el cambio en las reglas del juego (por ejemplo, el casino paga $\$2$ si el resultado es par y el jugador paga $\$1$ si es impar) a la ganancia esperada para el jugador después de 100 lanzamientos?

### Solucion Ejercicio 1

In [2]:
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd
import os
import scipy.stats as st

**Alumno 1 GS:**

In [1]:
# Solucion codigo

**Alumnos 2 EG:**

In [3]:
# Solucion codigo
n = 100000

dado1 = np.random.randint(1, 7, size = n)
dado2 = np.random.binomial(5, 0.5, size = n)
dado3 = np.random.binomial(5, 0.2, size = n)

suma = dado1 + dado2 + dado3

c7 = np.count_nonzero(suma == 7)
c14 = np.count_nonzero(suma == 14)
c18 = np.count_nonzero(suma == 18)

prob7 = c7 / n
prob14 = c14 / n
prob18 = c18 / n

prob7, prob14, prob18

(0.1598, 0.00038, 0.0)

La probabilidad de que sea _7_ es de 15.98%, de que sea _14_ es de 0.038% y de que salga _18_ es de 0%

### Solucion Ejercicio 2

**Alumno 1 GS:**

In [3]:
# Solucion codigo

**Alumno 2 EG:**

#### 1.

In [11]:
# Solucion codigo
from scipy.stats import binom

n = 100  
p = 0.5 
q = 1 - p
# Crear una distribución binomial
dist = binom(n, p)

# Calcular la probabilidad de que el jugador gane dinero
probabilidad_ganar = dist.sf(50)*100

print("La probabilidad de que el jugador gane dinero después de 100 lanzamientos es:", probabilidad_ganar)

La probabilidad de que el jugador gane dinero después de 100 lanzamientos es: 46.02053813064108


#### 2.

In [27]:
import math

def coeficiente_binomial(n, k):
    return math.factorial(n) // (math.factorial(k) * math.factorial(n - k))

def calcular_ganancia_esperada(p_par, p_impar, total_intentos, ganancia_par, ganancia_impar):
    ganancia_esperada = 0
    for i in range(total_intentos // 2 + 1):
        resultados_pares = i
        resultados_impares = total_intentos - i
        probabilidad = coeficiente_binomial(total_intentos, resultados_pares) * (p_par ** resultados_pares) * (p_impar ** resultados_impares)
        ganancia_esperada += probabilidad * (resultados_pares * ganancia_par - resultados_impares * ganancia_impar)
    return ganancia_esperada

# Nuevas reglas: casino paga $2 si es par, jugador paga $1 si es impar
p_par_nuevo = 1 / 2
p_impar_nuevo = 1 - p_par_nuevo
ganancia_par_nuevo = 2
ganancia_impar_nuevo = -1

total_intentos = 100

ganancia_esperada_nueva = calcular_ganancia_esperada(p_par_nuevo, p_impar_nuevo, total_intentos, ganancia_par_nuevo, ganancia_impar_nuevo)

print("Ganancia esperada con las nuevas reglas:", ganancia_esperada_nueva)


Ganancia esperada con las nuevas reglas: 78.97946186935894
